.github/workflows/test.yaml

```diff
@@ -0,0 +1,71 @@
+name: Test
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python     
+      #----------------------------------------------
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python 
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----      
+      #----------------------------------------------
+      - name: Install Poetry
+        uses: snok/install-poetry@v1.1.2
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+      #----------------------------------------------
+      #       load cached venv if cache exists      
+      #----------------------------------------------
+      - name: Load cached venv
+        id: cached-poetry-dependencies
+        uses: actions/cache@v2
+        with:
+          path: .venv
+          key: venv-${{ runner.os }}-${{ hashFiles('**/poetry.lock') }}
```

Comment:
The `poetry.lock` file is needed so this cache step can hash it and therefore restore the whole python environment based on it.

In the future when we want to update the lock file (and therefore all the dependencies of this project) we'll just run:

```
poetry update
```

And that will update the versions of all the packages in the lock file.  Then we can test the change of updating dependencies independent of testing other changes.  This way no upstream changes can surprise us and we can upgrade dependencies when we want to (also our CI is really fast because it just restores the whole python virtual environment).